Build FAILED on Win10 by msvc 2019 16.4.0

Build FAILED on Win 10 by msvc 2019 16.4.0, the error message follow:

6>Done building project "textures.vcxproj".
5>SurfaceOrientation.cpp
5>.\filament-1.4.3\libs\math\include\math/TVecHelpers.h(54,24): error C2938: 'std::enable_if_t<false,void>' : Failed to specialize alias template
5>.\filament-1.4.3\libs\math\include\math/TVecHelpers.h(54): message : see reference to alias template instantiation 'std::enable_if_t<false,void>' being compiled
5>.\filament-1.4.3\libs\math\include\math/vec2.h(96): message : see reference to alias template instantiation 'filament::math::details::enable_if_arithmetic_t<filament::math::half,int,int,int>' being compiled
5>.\filament-1.4.3\libs\math\include\math/vec2.h(101,15): error C2976: 'filament::math::vec2': too few template arguments
5>.\filament-1.4.3\libs\math\include\math/vec2.h(97): message : see declaration of 'filament::math::vec2'
5>.\filament-1.4.3\libs\math\include\math/vec3.h(121,15): error C2976: 'filament::math::vec3': too few template arguments
5>.\filament-1.4.3\libs\math\include\math/vec3.h(117): message : see declaration of 'filament::math::vec3'
5>.\filament-1.4.3\libs\math\include\math/vec4.h(119,15): error C2976: 'filament::math::vec4': too few template arguments
5>.\filament-1.4.3\libs\math\include\math/vec4.h(115): message : see declaration of 'filament::math::vec4'

--- a/libs/math/include/math/TVecHelpers.h
+++ b/libs/math/include/math/TVecHelpers.h
@@ -29,6 +29,20 @@
 
 namespace filament {
 namespace math {
+
+// MSVC 2019 16.4 doesn't seem to like it when we specialize std::is_arithmetic for
+// filament::math::half, so we're forced to create own is_arithmetic here and specialize it inside
+// of half.h.
+template<typename T>
+struct is_arithmetic : std::integral_constant<bool,
+        std::is_integral<T>::value || std::is_floating_point<T>::value> {
+};
+
+}
+}
+
+namespace filament {
+namespace math {
 namespace details {
 // -------------------------------------------------------------------------------------
 
@@ -52,10 +66,10 @@ using arithmetic_result_t = typename arithmetic_result<T, U>::type;
 
 template<typename A, typename B = int, typename C = int, typename D = int>
 using enable_if_arithmetic_t = std::enable_if_t<
-        std::is_arithmetic<A>::value &&
-        std::is_arithmetic<B>::value &&
-        std::is_arithmetic<C>::value &&
-        std::is_arithmetic<D>::value>;
+        is_arithmetic<A>::value &&
+        is_arithmetic<B>::value &&
+        is_arithmetic<C>::value &&
+        is_arithmetic<D>::value>;
 
 /*
  * No user serviceable parts here.

--- a/libs/math/include/math/TVecHelpers.h
+++ b/libs/math/include/math/TVecHelpers.h
@@ -29,20 +29,6 @@
 
 namespace filament {
 namespace math {
-
-// MSVC 2019 16.4 doesn't seem to like it when we specialize std::is_arithmetic for
-// filament::math::half, so we're forced to create own is_arithmetic here and specialize it inside
-// of half.h.
-template<typename T>
-struct is_arithmetic : std::integral_constant<bool,
-        std::is_integral<T>::value || std::is_floating_point<T>::value> {
-};
-
-}
-}
-
-namespace filament {
-namespace math {
 namespace details {
 // -------------------------------------------------------------------------------------
 

--- a/libs/math/include/math/compiler.h
+++ b/libs/math/include/math/compiler.h
@@ -16,6 +16,8 @@
 
 #pragma once
 
+#include <type_traits>
+
 #if defined (WIN32)
 
 #ifdef max
@@ -108,3 +110,17 @@
 #   define CONSTEXPR_IF_NOT_MSVC constexpr
 
 #endif // _MSC_VER
+
+namespace filament {
+namespace math {
+
+// MSVC 2019 16.4 doesn't seem to like it when we specialize std::is_arithmetic for
+// filament::math::half, so we're forced to create our own is_arithmetic here and specialize it
+// inside of half.h.
+template<typename T>
+struct is_arithmetic : std::integral_constant<bool,
+        std::is_integral<T>::value || std::is_floating_point<T>::value> {
+};
+
+}
+}

--- a/libs/math/include/math/half.h
+++ b/libs/math/include/math/half.h
@@ -22,7 +22,6 @@
 #include <type_traits>
 
 #include <math/compiler.h>
-#include <math/TVecHelpers.h>
 
 namespace filament {
 namespace math {

--- a/libs/math/include/math/half.h
+++ b/libs/math/include/math/half.h
@@ -22,6 +22,7 @@
 #include <type_traits>
 
 #include <math/compiler.h>
+#include <math/TVecHelpers.h>
 
 namespace filament {
 namespace math {
@@ -162,6 +163,8 @@ inline constexpr half operator "" _h(long double v) {
     return half( static_cast<float>(v) );
 }
 
+template<> struct is_arithmetic<filament::math::half> : public std::true_type {};
+
 } // namespace math
 } // namespace filament
 
@@ -170,6 +173,8 @@ namespace std {
 template<> struct is_floating_point<filament::math::half> : public std::true_type {};
 
 // note: this shouldn't be needed (is_floating_point<> is enough) but some version of msvc need it
+// This stopped working with MSVC 2019 16.4, so we specialize our own version of is_arithmetic in
+// the math::filament namespace (see above).
 template<> struct is_arithmetic<filament::math::half> : public std::true_type {};
 
 template<>

